core/iwasm/aot/arch/aot_reloc_riscv.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2021 XiaoMi Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "aot_reloc.h"
+
+#define R_RISCV_CALL     18
+#define R_RISCV_CALL_PLT 19
+#define R_RISCV_HI20     26
+#define R_RISCV_LO12_I   27
+#define R_RISCV_LO12_S   28
+
+#define RV_OPCODE_SW 0x23
+
+static SymbolMap target_sym_map[] = { REG_COMMON_SYMBOLS };
+
+void
+get_current_target(char *target_buf, uint32 target_buf_size)
+{
+    snprintf(target_buf, target_buf_size, "riscv");
+}
+
+uint32
+get_plt_item_size()
+{
+    return 16;
+}
+
+SymbolMap *
+get_target_symbol_map(uint32 *sym_num)
+{
+    *sym_num = sizeof(target_sym_map) / sizeof(SymbolMap);
+    return target_sym_map;
+}
+
+/* Get a val from given address */
+static uint32
+rv_get_val(uint16_t *addr)
+{
+    uint32_t ret;
+    ret = *addr | (*(addr + 1)) << 16;
```

Comment:
Should we consider little endian and big endian? In aot_reloc_xtensa.c, the is_little_endian() macro is defined as:
```C
static union {
    int a;
    char b;
} __ue = { .a = 1 };

#define is_little_endian() (__ue.b == 1)
```